### PR TITLE
Added diagram and fixed urls.

### PR DESCRIPTION
**Pull request by Meruja**

### CODE REVIEW

File: en/docs/learn/api-security/oauth2/grant-types/authorization-code-grant.md [hunk 1]
```diff
@@ -2,53 +2,59 @@
...
 
 The client then requests an access token from the authorization server's `/token` endpoint by including the authorization code received in the previous step. When making the request, the client authenticates with the authorization server. It then includes the redirection URI used to obtain the authorization code for verification. The authorization server authenticates the client, validates the authorization code, and ensures that the redirection URI matches the URI used to redirect the client from the /authorize endpoint in the previous response. If valid, the authorization server responds back with an access token and, optionally, a refresh token.
 
+The diagram below depicts the flow of Authorization Code Grant :
+
+![](../../../../assets/img/learn/oauth-authorization-code-grant-diagram.png)
```
Comment (vithu30) [inline en/docs/learn/api-security/oauth2/grant-types/authorization-code-grant.md]: Shall we replace it with basepath?

File: en/docs/learn/api-security/oauth2/grant-types/authorization-code-grant.md [hunk 2]
```diff
@@ -63,52 +69,67 @@ The steps below show how access tokens are generated for the authorization code
...
+1.  Open a terminal window and add the following entry to the `/etc/hosts` file of your machine to configure the hostname.
+    ```
+    127.0.0.1   wso2is.local
+    127.0.0.1   localhost.com
+    ```
+        
+    !!!info "Why is this step needed?"
+        Some browsers do not allow you to create cookies for a naked hostname, such as `localhost`. Cookies are required when working with SSO. Therefore, to ensure that the SSO capabilities work as expected in this tutorial, you need to configure the `/etc/host` file as explained in this step.
+            
+        The `/etc/host` file is a read-only file. Therefore, you won't be able to edit it by opening the file via a text editor. Instead, edit the file using the terminal commands. For example, use the following command if you are working on a Mac/Linux environment.
```
Comment (vithu30) [inline en/docs/learn/api-security/oauth2/grant-types/authorization-code-grant.md]: I think the file name should be changed as `/etc/hosts`?

Comment (Meruja) [reply en/docs/learn/api-security/oauth2/grant-types/authorization-code-grant.md]: fixed with 34f4e03

File: en/docs/learn/api-security/oauth2/grant-types/authorization-code-grant.md (hunk 1 above)
Comment (Meruja) [reply en/docs/learn/api-security/oauth2/grant-types/authorization-code-grant.md]: fixed with 34f4e03

**Comment by Mariangela** [comment]
@Meruja , can you check and resolve the conflicts in the PR.

**Comment by Mariangela** [comment]
@Meruja , is this change only applicable for 3.1.0?

**Comment by Meruja** [comment]
> @Meruja , is this change only applicable for 3.1.0?

Applicable for 3.0.0 as well

**Pull request merged**